test/model/src/include/model/driver/kv_workload_runner_wt.h

```diff
@@ -0,0 +1,345 @@
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef MODEL_DRIVER_KV_WORKLOAD_RUNNER_WT_H
+#define MODEL_DRIVER_KV_WORKLOAD_RUNNER_WT_H
+
+#include <memory>
+#include <shared_mutex>
+#include <unordered_map>
+#include "model/driver/kv_workload.h"
+#include "model/core.h"
+#include "wiredtiger.h"
+
+namespace model {
+
+/*
+ * kv_workload_runner_wt --
+ *     The workload context for WiredTiger.
+ */
+class kv_workload_runner_wt {
+
+protected:
+    /*
+     * session_context --
+     *     The WiredTiger session
+     */
+    class session_context {
+
+        using cursor_id_t = int;
+        static constexpr cursor_id_t k_cursors_per_table = 16;
+
+    public:
+        /*
+         * session_context::session_context --
+         *     Create the
+         */
+        inline session_context(kv_workload_runner_wt &workload_context, WT_SESSION *session)
+            : _session(session), _workload_context(workload_context)
+        {
+        }
+
+        /*
+         * session_context::~session_context --
+         *     Destroy the context, alongside the corresponding resources.
+         */
+        ~session_context();
+
+        /*
+         * session_context::session --
+         *     Get the session.
+         */
+        inline WT_SESSION *
+        session() const noexcept
+        {
+            return _session;
+        }
+
+        /*
+         * session_context::cursor --
+         *     Get a cursor. Create one if it does not already exist. Use the second argument to get
+         *     and/or create additional cursors for the given table.
+         */
+        WT_CURSOR *cursor(table_id_t table_id, int table_cur_id = 0);
+
+    private:
+        /*
+         * session_context::cursor_id --
+         *     Get a cursor ID.
+         */
+        static inline cursor_id_t
+        cursor_id(table_id_t table_id, int table_cur_id)
```

Comment:
Does `table_cur_id` need to be an `int` instead of `unsigned`? The same question applies to `table_id_t`. Is it because a negative value would be considered as unset?

Reply:
Not really; I changed it to `unsigned` as suggested. Bad habit from many years of Java programming.